modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataMediator.java

```diff
@@ -148,10 +122,22 @@ public MetadataMediator(ComponentModel componentModel) {
     return mergeResults(componentDescriptor, output, input);
   }
 
+  private MetadataResult<Object> getMetadataKeyObjectValue(ParameterValueResolver metadataKeyResolver) {
+    try {
+      Object keyValue = getContainerName().isPresent() ? metadataKeyResolver.getParameterValue(getContainerName().get()) : null;
+      return success(keyValue);
```

Comment:
why wrapping this in a metadata result? I'd just handle the exception outside and generate the failure there if needed.
regarding the `null`, I get that in the keyless case we have to pass on something to the resolver, but I'm not fond on injecting null to a piece of code that might be reused in cases where that key actually has a value. 
maybe we can think of a way of handling this cases from the resolver and all the way up to here (for example, if a resolver is used in a keyless way, we force them to parameterize the resolver with an Optional<> key, and we wrap whatever comes here in an optional.ofNullable to handle both cases)